### PR TITLE
feat(SidePanel): allow actionToolbarButtons to be a bit more flexible

### PR DESCRIPTION
**Pull request by matthewgallo**
Contributes to #1177 

Adds some flexibility to the action toolbar buttons in the side panel and properly displays the tooltip via `hasIconOnly` prop on buttons.

#### What did you change?
`SidePanel.js`
`SidePanel.test.js`
`SidePanel.stories.js`
#### How did you test and verify your work?
Storybook and updated tests

### CODE REVIEW

File: packages/cloud-cognitive/src/components/SidePanel/SidePanel.js [hunk 1]
```diff
@@ -505,23 +507,25 @@ export let SidePanel = React.forwardRef(
...
                     [`${blockClass}__action-toolbar-icon-only-button`]:
-                      action.icon,
+                      action.icon && !action.leading,
                     [`${blockClass}__action-toolbar-leading-button`]:
-                      !action.icon,
+                      action.leading,
                   },
                 ])}
-                onClick={() => action.onActionToolbarButtonClick()}>
+                onClick={() => action.onClick()}>
```
Comment (SimonFinney) [inline packages/cloud-cognitive/src/components/SidePanel/SidePanel.js:528]: This makes sense for both consistency and predictability — will it break existing consumers?

One other thought, and I likely need a history lesson on this, but is there a rationale for preventing the event parameter to be passed to the callback?

**Review by SimonFinney — COMMENTED** [review]
Thanks @matthewgallo — this makes absolute sense to me but I wanted to clarify something first. 🙏

File: packages/cloud-cognitive/src/components/SidePanel/SidePanel.js (hunk 1 above)
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/SidePanel/SidePanel.js]: Updated @SimonFinney

Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/SidePanel/SidePanel.js]: Screen Shot 2021-08-23 at 6 07 40 PM

Used the `deprecateProp` helper that will give a console warning if `onActionToolbarButtonClick` is used.

**Review by SimonFinney — APPROVED** [review]
LGTM — thank you! 💯

**Pull request merged**